pipeline/submodules/select_direction.py

```diff
@@ -114,6 +114,137 @@ def filter_fn(refusal_score, steering_score, kl_div_score, layer, n_layer, kl_th
+            ablation_refusal_scores[source_pos, source_layer] = refusal_scores.mean().item()
+
+    for source_pos in range(-1, 0):
+        for source_layer in tqdm(range(n_layer), desc=f"Computing refusal addition for source position {source_pos}"):
+
+            refusal_vector = direction_to_add
+            coeff = torch.tensor(1.0)
+
+            fwd_pre_hooks = [(model_base.model_block_modules[source_layer], get_activation_addition_input_pre_hook(vector=refusal_vector, coeff=coeff))]
+            fwd_hooks = []
+
+            refusal_scores = get_refusal_scores(model_base.model, harmless_instructions, model_base.tokenize_instructions_fn, model_base.refusal_toks, fwd_pre_hooks=fwd_pre_hooks, fwd_hooks=fwd_hooks, batch_size=batch_size)
+            steering_refusal_scores[source_pos, source_layer] = refusal_scores.mean().item()
+
+    plot_refusal_scores(
+        refusal_scores=ablation_refusal_scores,
+        baseline_refusal_score=baseline_refusal_scores_harmful.mean().item(),
+        token_labels=model_base.tokenizer.batch_decode(model_base.eoi_toks),
+        title='Ablating direction on harmful instructions',
+        artifact_dir=artifact_dir,
+        artifact_name='ablation_scores'
+    )
+
+    plot_refusal_scores(
+        refusal_scores=steering_refusal_scores,
+        baseline_refusal_score=baseline_refusal_scores_harmless.mean().item(),
+        token_labels=model_base.tokenizer.batch_decode(model_base.eoi_toks),
+        title='Adding direction on harmless instructions',
+        artifact_dir=artifact_dir,
+        artifact_name='actadd_scores'
+    )
+
+    plot_refusal_scores(
+        refusal_scores=ablation_kl_div_scores,
+        baseline_refusal_score=0.0,
+        token_labels=model_base.tokenizer.batch_decode(model_base.eoi_toks),
+        title='KL Divergence when ablating direction on harmless instructions',
+        artifact_dir=artifact_dir,
+        artifact_name='kl_div_scores'
+    )
+
+    filtered_scores = []
+    json_output_all_scores = []
+    json_output_filtered_scores = []
+
+    for source_pos in range(-1, 0):
+        for source_layer in range(n_layer):
+
+            json_output_all_scores.append({
+                'position': source_pos,
+                'layer': source_layer,
+                'refusal_score': ablation_refusal_scores[source_pos, source_layer].item(),
+                'steering_score': steering_refusal_scores[source_pos, source_layer].item(),
+                'kl_div_score': ablation_kl_div_scores[source_pos, source_layer].item()
+            })
+
+            refusal_score = ablation_refusal_scores[source_pos, source_layer].item()
+            steering_score = steering_refusal_scores[source_pos, source_layer].item()
+            kl_div_score = ablation_kl_div_scores[source_pos, source_layer].item()
+
+    with open(f"{artifact_dir}/direction_evaluations.json", 'w') as f:
+        json.dump(json_output_all_scores, f, indent=4)
+
+    return -2, 10, direction_to_add
```

Comment:
This is a useless return statement. You can just remove it or return something useful.